test/unit/test_borealis.py

```diff
@@ -0,0 +1,1031 @@
+        """
+        self.assertRaises(OSError, pydarn.BorealisSiteRead,
+                          self.nonexistent_file_path, 'rawacf')
+
+    def test_empty_file(self):
+        """
+        Tests if BorealisSiteRead constructor with an empty file
+
+        Expected behaviour: raise OSError, unable to open (file signature not found)
+        """
+        self.assertRaises(OSError,
+                          pydarn.BorealisSiteRead, self.empty_file_path, 'rawacf')
+
+    def test_read_bfiq(self):
+        """
+        Test reading records from bfiq.
+
+        Checks:
+            - returns correct data structures
+            - returns expected values
+        """
+        dm = pydarn.BorealisSiteRead(self.bfiq_file_path, 'bfiq')
+        records = dm.records
+        first_record = records[dm.record_names[0]]
+        self.assertIsInstance(records, collections.OrderedDict)
+        self.assertIsInstance(first_record, dict)
+        self.assertIsInstance(first_record['num_slices'], np.int64)
+        del dm, records
+
+    def test_read_rawacf(self):
+        """
+        Test reading records from rawacf.
+
+        Checks:
+            - returns correct data structures
+            - returns expected values
+        """
+        dm = pydarn.BorealisSiteRead(self.rawacf_file_path, 'rawacf')
+        records = dm.records
+        first_record = records[dm.record_names[0]]
+        self.assertIsInstance(records, collections.OrderedDict)
+        self.assertIsInstance(first_record, dict)
+        self.assertIsInstance(first_record['num_slices'], np.int64)
+        del dm, records
+
+    def test_read_antennas_iq(self):
+        """
+        Test reading records from antennas_iq.
+
+        Checks:
+            - returns correct data structures
+            - returns expected values
+        """
+        dm = pydarn.BorealisSiteRead(self.antennas_file_path, 'antennas_iq')
+        records = dm.records
+        first_record = records[dm.record_names[0]]
+        self.assertIsInstance(records, collections.OrderedDict)
+        self.assertIsInstance(first_record, dict)
+        self.assertIsInstance(first_record['num_slices'], np.int64)
+        del dm, records
+
+    # TODO add rawrf file for test
+    # def test_read_rawrf(self):
+    #     """
+    #     Test reading records from antennas_iq.
+
+    #     Checks:
+    #         - returns correct data structures
+    #         - returns expected values
+    #     """
+    #     dm = pydarn.BorealisSiteRead(self.rawrf_file_path, 'rawrf')
+    #     records = dm.records
+    #     first_record = records[dm.record_names[0]]
+    #     self.assertIsInstance(records, collections.OrderedDict)
+    #     self.assertIsInstance(first_record, dict)
+    #     self.assertIsInstance(first_record['num_slices'], np.int64)
+
+
+class TestBorealisSiteWrite(unittest.TestCase):
+    """
+    Tests BorealisSiteWrite class
+    """
+
+    def setUp(self):
+        self.rawacf_site_data = copy.deepcopy(
```

Comment:
Why is deepcopy being used?

Reply:
Pointers :)
When you import data from a class and want to reuse it you have to make a deep copy or else your changes will stay with it throughout the program known as a shallow copy. 

https://www.geeksforgeeks.org/copy-python-deep-copy-shallow-copy/